app/assets/javascripts/taxon-path-prefixer.js

```diff
@@ -0,0 +1,47 @@
+(function (Modules) {
+  "use strict";
+
+  Modules.TaxonPathPrefixer = function () {
+    this.start = function() {
+      var $parentSelectEl = $('select.js-parent-taxon');
+      var $pathPrefixInputEl = $('input.js-path-prefix');
+      var $pathSlugInputEl = $('input.js-path-slug');
+      var $previewBasePathEl = $('.js-predicted-base-path');
+
+      // When editing a root taxon, allow the "prefix"
+      // value to be edited via the empty slug field
+      if ($parentSelectEl.val().length === 0) {
+        var pathPrefix = $pathPrefixInputEl.val();
+
+        $pathSlugInputEl.val(pathPrefix);
+        $pathPrefixInputEl.val('');
+
+        $previewBasePathEl.text(pathPrefix);
+      }
+
+      updateBasePath();
+      $parentSelectEl.change(updateBasePath);
+
+      var timeout = null;
+      $pathSlugInputEl.on('keyup', function() {
+        clearTimeout(timeout);
+        timeout = setTimeout(updateBasePath, 500);
```

Comment:
Just curious about this timeout... why did you go for 500 or a timeout in general?